Gems/Atom/RHI/Vulkan/Code/Source/RHI/PhysicalDevice.cpp

```diff
@@ -262,6 +262,7 @@ namespace AZ
             } };
 
             uint32_t optionalExtensionCount = sizeof(optionalExtensions) / sizeof(VK_EXT_SAMPLE_LOCATIONS_EXTENSION_NAME);
+            AZ_UNUSED(optionalExtensionCount); // optionalExtensionCount is only used in asserts.
```

Comment:
here you can do:
```
#if defined(AZ_ENABLE_TRACING)
uint32_t optionalExtensionCount = sizeof(optionalExtensions) / sizeof(VK_EXT_SAMPLE_LOCATIONS_EXTENSION_NAME);
#endif
```
That will be a bit better to avoid the computation of that variable when AZ_Assert is not needed.
You can also do:
```
[[maybe_unused]] uint32_t optionalExtensionCount = sizeof(optionalExtensions) / sizeof(VK_EXT_SAMPLE_LOCATIONS_EXTENSION_NAME);
```